packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -156,23 +156,30 @@ class BuildableMacOSApp extends MacOSApp {
   String get name => 'macOS';
 
   @override
-  String? applicationBundle(BuildMode buildMode) {
+  String? applicationBundle(BuildInfo buildInfo) {
     final File appBundleNameFile = project.nameFile;
     if (!appBundleNameFile.existsSync()) {
       globals.printError('Unable to find app name. ${appBundleNameFile.path} does not exist');
       return null;
     }
+
+    String bundleDirectory = sentenceCase(buildInfo.mode.name);
+    final String? flavor = buildInfo.flavor;
+    if (flavor != null) {
+      bundleDirectory += ' ${sentenceCase(flavor)}';
+    }
```

Comment:
Is there a test for this?

Reply:
There is now https://github.com/flutter/flutter/pull/118421/commits/8b25cb74e0f894ed4299db387cdfae563766436d 👍